internal/ingress/controller/reconciler.go

```diff
@@ -0,0 +1,87 @@
+package controller
+
+import (
+	"context"
+	"github.com/kubernetes-sigs/aws-alb-ingress-controller/internal/alb/lb"
+	"github.com/kubernetes-sigs/aws-alb-ingress-controller/internal/ingress/controller/store"
+	"github.com/kubernetes-sigs/aws-alb-ingress-controller/internal/ingress/metric"
+	extensions "k8s.io/api/extensions/v1beta1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/client-go/tools/record"
+	"sigs.k8s.io/controller-runtime/pkg/cache"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+)
+
+// Reconciler reconciles an single ingress object
+type Reconciler struct {
+	client   client.Client
+	cache    cache.Cache
+	recorder record.EventRecorder
+
+	// TODO: move things out of store, and start to rely on functionality provided by client & cache
+	store store.Storer
+
+	lbController lb.LoadBalancerController
+
+	metricCollector metric.Collector
+}
+
+func (r *Reconciler) Reconcile(request reconcile.Request) (reconcile.Result, error) {
+	instance := &extensions.Ingress{}
+
+	ctx := context.Background()
+	err := r.cache.Get(ctx, request.NamespacedName, instance)
+	if err != nil {
+		if !errors.IsNotFound(err) {
+			r.metricCollector.IncReconcileErrorCount(request.NamespacedName.String())
+			return reconcile.Result{}, err
+		}
+
+		if err := r.deleteIngress(ctx, request.NamespacedName); err != nil {
```

Comment:
if the delete fails, would it ever be tried again?

Reply:
yes, if we return an err, it will be retried in backoff manner. 😄